tests/ui/incorrect_partial_ord_impl_on_ord_type.rs

```diff
@@ -0,0 +1,117 @@
+#![allow(unused)]
+#![no_main]
+
+use std::cmp::Ordering;
+
+// lint
+
+#[derive(Eq, PartialEq)]
+struct A(u32);
+
+impl Ord for A {
+    fn cmp(&self, other: &Self) -> Ordering {
+        todo!();
+    }
+}
+
+impl PartialOrd for A {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        todo!();
+    }
+}
+
+// do not lint
+
+#[derive(Eq, PartialEq)]
+struct B(u32);
+
+impl Ord for B {
+    fn cmp(&self, other: &Self) -> Ordering {
+        todo!();
+    }
+}
+
+impl PartialOrd for B {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+// lint, but we can't give a correct suggestion since &Self is not named
```

Comment:
I think it would be valid to just add a second suggestion, giving `&Self` a name. Then we can also test the suggestion in the CI. Otherwise, I would suggest splitting the file into `_fixable.rs` and a `unfixable.rs` version :)